File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) return false;
        vector<int> myvec; vector<int> revvec;
        int num = x;
        while(num > 0){
            myvec.insert(myvec.begin(), num%10);
            revvec.push_back(num%10);
            num = num/10;
        }
        if (myvec == revvec) return true;
        else return false;
    };
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {

        if (x < 0 || (x % 10 == 0 && x != 0)) {
            return false;
        }

        if (x >= 0 && x < 10) {
            return true;
        }

        int revHalf = 0;

        while (x > revHalf) {
            revHalf = revHalf * 10 + x % 10;
            x /= 10;
        }

        return x == revHalf || x == revHalf / 10;
    }
}